event/handler_test.go

```diff
@@ -1,29 +1,62 @@
 package event_test
 
 import (
+	"context"
+	"errors"
 	"os"
 	"testing"
 
 	"github.com/ONSdigital/dp-search-data-extractor/config"
 	"github.com/ONSdigital/dp-search-data-extractor/event"
+	eventMock "github.com/ONSdigital/dp-search-data-extractor/event/mock"
 	. "github.com/smartystreets/goconvey/convey"
 )
 
-// TODO: remove hello called example test
-func TestHelloCalledHandler_Handle(t *testing.T) {
+var errZebedee = errors.New("zebedee test error")
+var getPublishDataFuncInError = func(ctx context.Context, uriString string) ([]byte, error) {
+	return nil, errZebedee
+}
+
+var getPublishDataFunc = func(ctx context.Context, uriString string) ([]byte, error) {
+	data := []byte("test data")
+	return data, nil
+}
+
+func TestContentPublishedHandler_Handle(t *testing.T) {
 
 	Convey("Given a successful event handler, when Handle is triggered", t, func() {
```

Comment:
This test is not very readable in my view. It seems to be missing a "Then" statement. I would suggest separating the "When" out from the "Given" statement too. Something like this: 
```suggestion
	Convey("Given an event handler, working successfully, and an event containing a URL", t, func() {
		var zebedeeMock = &eventMock.ZebedeeClientMock{GetPublishedDataFunc: getPublishDataFunc}
		eventHandler := &event.ContentPublishedHandler{zebedeeMock}
		filePath := "/tmp/dpSearchDataExtractor.txt"
		os.Remove(filePath)

		Convey("When the event handler is triggered", func() {
			err := eventHandler.Handle(testCtx, &config.Config{OutputFilePath: filePath}, &testEvent)

			Convey("Then the published data endpoint should be called, using the URL, and no error returned", func() {
				So(err, ShouldBeNil)
				So(zebedeeMock.GetPublishedDataCalls(), ShouldNotBeEmpty)
				So(zebedeeMock.GetPublishedDataCalls(), ShouldHaveLength, 1)
				So(zebedeeMock.GetPublishedDataCalls()[0].UriString, ShouldEqual, testEvent.URL)
			})
		})
	})
```